pkg/zfs/mount.go

```diff
@@ -69,13 +76,16 @@ func UmountVolume(vol *apis.ZFSVolume, targetPath string,
 		return err
 	}
 
-	if err = SetDatasetLegacyMount(vol); err != nil {
-		// ignoring the failure as the volume has already
-		// been umounted, now the new pod can mount it
-		logrus.Warningf(
-			"zfs: failed to set legacy mountpoint: %s err: %v",
-			vol.Name, err,
-		)
+	// TODO: ignore the legacy mount feature for a moment until correctly sharing
```

Comment:
This is the part I temporarily disabled. Sorry about that, just wanted the sharing to work. Let's talk about options later on.